SayTheirNames/Source/Controller/Donations/DonationsMoreDetails/DonationsMoreDetailsController.swift

```diff
@@ -176,14 +230,20 @@ final class DonationsMoreDetailsController: UIViewController {
     private func setupDonationButton() {
         donationButtonContainerView.setButtonPressed { [weak self] in
             guard let self = self else { return }
-            guard let donationURL = URL(string: self.donation.link) else {
+            guard let donationURL = URL(string: self.data.entity.link) else {
                        assertionFailure("Invalid donationURL")
                        return
                    }
             UIApplication.shared.open(donationURL)
         }
         
-        donationButtonContainerView.setButtonTitle(Strings.donateNow)
+        switch data {
+        case .donation:
+            donationButtonContainerView.setButtonTitle(L10n.donateNow)
+        case .petition:
+            donationButtonContainerView.setButtonTitle("Sign") // TODO: localize
```

Comment:
Any reason we just didnt localize this?

Reply:
localisation is broken at the moment so can't add new ones. it's a follow-up task

Reply:
Nooooooo :(